tests/main/test_task.py

```diff
@@ -779,3 +786,192 @@ def llm_response(
 
     result = processor_task[int].run(10)
     assert result == 11
+
+
+def test_task_output_format_sequence(
+    test_settings: Settings,
+):
+    """
+    Test that `Task`s correctly execute a sequence of steps
+    controlled by the agent's `output_format`, and that `output_format`
+    is handled by default without `enable_message`.
+    """
+
+    set_global(test_settings)
+
+    class MultiplyTool(ToolMessage):
+        request: str = "multiply"
+        purpose: str = "To multiply two integers."
+        a: int
+        b: int
+
+    class IncrementTool(ToolMessage):
+        request: str = "increment"
+        purpose: str = "To increment an integer."
+        x: int
+
+    class PowerTool(ToolMessage):
+        request: str = "power"
+        purpose: str = "To compute `x` ** `y`."
+        x: int
+        y: int
+
+    class CompositionAgent(ChatAgent):
+        def __init__(self, config: ChatAgentConfig = ChatAgentConfig()):
+            super().__init__(config)
+            self.set_output_format(MultiplyTool)
+
+        def multiply(self, message: MultiplyTool) -> str:
+            self.set_output_format(IncrementTool)
+
+            return str(message.a * message.b)
+
+        def increment(self, message: IncrementTool) -> str:
+            self.set_output_format(PowerTool)
+
+            return str(message.x + 1)
+
+        def power(self, message: PowerTool) -> str:
+            self.set_output_format(MultiplyTool)
```

Comment:
we don't need to set this I think? since power is the last step

Reply:
I was thinking that we should set up a cycle to allow agent reuse; this isn't needed here, so I'll remove.